### ISSUE TITLE
Exclude from Proxy doesn't exclude from HTTP Sessions "Site" pull down choices

### ISSUE THREAD

**Issue by zestysoft** [zaproxy/zaproxy#5275]
**Describe the bug**
After removing a number of sites by right clicking and choosing "Exclude From..., Proxy", they continue to exist as a choice when pulling down "Site" in the HTTP Sessions tab.

**To Reproduce**

**Expected behavior**
Sites should be filtered out here too.

**Screenshots**

**Software versions**
ZAP Weekly

**Comment by kingthorin** [comment]
Do they if you delete them from the sites tree?

Technically the Sessions tab isn’t the proxy, “Exclude From Proxy” is a totally literal title. They’re still applicable for other tools (spider, active scan, etc).

**Comment by zestysoft** [comment]
The right-click exclude from.., Proxy is being done from the Sites tree.  They vanish from the sites tree after I do this.

**Comment by kingthorin** [comment]
Yup, but if you make a manual request (request editor) etc it will reappear, it simply isn't applicable to proxy functionality. 

You may still care about the sessions if you spider or active scan it. However, I guess the argument could be made that it shouldn't be removed from Sites Tree/History, just that further interactions shouldn't be proxied (added to Sites Tree/History)...

**Comment by zestysoft** [comment]
Ah okay, so that brings up some other false assumptions I made that got me to incorrectly use this feature.  I'm going to continue this conversation back in the user group and explain further.

**Comment by zestysoft** [comment]
Okay, so it looks like this is a bug after-all.  Deleting the nodes from Sites doesn't remove the entries from the sites pull down either.

Also, I think the scope of this problem is larger, however this might be considered a different bug.

After using the global ignore list, when I open Firefox, nothing appears under Sites as expected, however, something does appear under HTTP Sessions:
![Screenshot from 2019-03-22 11-47-17](https://user-images.githubusercontent.com/742554/54846094-4f425d00-4c98-11e9-9957-c5efec84e7f9.png)

Oddly, github.com isn't in my ignore list -- so I would expect something to show up under sites -- I assume this is related to the Zap HUD code?

![Screenshot from 2019-03-22 11-52-12](https://user-images.githubusercontent.com/742554/54846399-f9ba8000-4c98-11e9-99d9-06c0310f54eb.png)

**Comment by kingthorin** [comment]
@zestysoft Thanks for the follow-up. As you might imagine things have been built over the last 9 years by people all over the globe and there are bound to be interactions like these (that you've outlined) that get missed. Thanks for the catch(es)!
